extensions/SharkPool/Recording.js

```diff
@@ -0,0 +1,285 @@
+            text: "average loudness"
+          },
+          "---",
+          {
+            opcode: "whenMic",
+            blockType: Scratch.BlockType.EVENT,
+            text: "when mic turns [ON_OFF]",
+            isEdgeActivated: false,
+            arguments: {
+              ON_OFF: { type: Scratch.ArgumentType.STRING, menu: "ACTIVE" }
+            }
+          },
+          {
+            opcode: "isRecordingMic",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "is recording?"
+          },
+          "---",
+          {
+            opcode: "recordedAudio",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "recorded audio as [TYPE]",
+            arguments: {
+              TYPE: { type: Scratch.ArgumentType.STRING, menu: "FILETYPES" }
+            }
+          },
+          {
+            opcode: "saveRecording",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "save recording to [SPRITE] named [NAME]",
+            arguments: {
+              SPRITE: { type: Scratch.ArgumentType.STRING, menu: "TARGETS" },
+              NAME: { type: Scratch.ArgumentType.STRING, defaultValue: "Recording 1" }
+            }
+          },
+          {
+            opcode: "saveRecording2",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "download recording named [NAME] as [TYPE]",
+            arguments: {
+              NAME: { type: Scratch.ArgumentType.STRING, defaultValue: "Recording_1" },
+              TYPE: { type: Scratch.ArgumentType.STRING, menu: "FILETYPES" }
+            }
+          }
+        ],
+        menus: {
+          TARGETS: { acceptReporters: true, items: "_getTargets" },
+          ACTIVE: ["on", "off"],
+          MODE: { acceptReporters: true, items: ["enabled", "disabled"] },
+          FILETYPES: {
+            acceptReporters: true,
+            items: ["mp3", "wav", "mpeg", "ogg"]
+          }
+        },
+      };
+    }
+
+    // Helper Funcs
+    _getTargets() {
+      const spriteNames = [];
+      const targets = runtime.targets;
+      for (let index = 0; index < targets.length; index++) {
+        const target = targets[index];
+        if (target.isOriginal) spriteNames.push(target.getName());
+      }
+      return spriteNames.length > 0 ? spriteNames : [""];
+    }
+
+    blob2Base64(blob, TYPE) {
+      return new Promise((resolve) => {
+        const reader = new FileReader();
+        reader.onloadend = () => {
+          const base64 = reader.result.split(",")[1];
+          resolve(`data:audio/${TYPE};base64,${base64}`);
+        };
+        reader.onerror = () => { resolve("") };
+        reader.readAsDataURL(blob);
+      });
+    }
+
+    allowAccess() {
+      return window.confirm("Allow access to record Microphone Audio? Be aware of privacy concerns if you Accept.");
+    }
+
+    // Block Funcs
+    recordingSet(args, util) {
+      if (args.MODE === "disabled") this.stopRecording();
+      else {
```

Comment:
```suggestion
      else if (args.MODE === "enabled") {
```
Since this block accepts reporters, you should be checking if the argument is "enabled", rather than checking for the argument not being "disabled".